### PR TITLE
chore: Improve the documentation regarding the selection of Application on by Sync Windows (#20971)

### PR DESCRIPTION
**Pull request by thecooldrop**
Added documentation requested by issue #20971

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates? Yes
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Review by reggie-k — APPROVED** [review]

File: docs/user-guide/sync_windows.md [hunk 1]
```diff
@@ -2,7 +2,31 @@
...
+are three mechanisms for selecting the Application resources to which a Sync Window applies:
+
+- By name of Application resource
+- By cluster into which resources are installed by Application resource. This is specified by `Application.spec.destination.name` and `.server` fields
+- By namespace into which resources are installed by Application resource. This is specified by `Application.spec.destination.namespace` field.
+
+All three mechanisms allow usage of wildcards. The mechanisms are not mutually exclusive, and all three of them can be used in single
+Sync Window definition. 
+
+When multiple selection mechanisms are used, they are effectively `ORed`, meaning that if any of the selector selects the Application,
```
Comment (adriananeci) [inline docs/user-guide/sync_windows.md:24]: I've raised a while ago a PR for adding also `AND` support. https://github.com/argoproj/argo-cd/pull/16846 is still open, not sure if you want to mentioned it here or I can update the docs once that PR is close to being merged.

**Review by nitishfy — APPROVED** [review]
LGTM!

File: docs/user-guide/sync_windows.md (hunk 1 above)
Comment (vaniobegic) [reply docs/user-guide/sync_windows.md:24]: Hi @adriananeci, I think that proper way forward here is to have your branch rebased onto main after the documentation here is merged, and then to update the documentation in your branch. 

If we change the documentation now, we will be adding documentation for features, which are not yet included in the mainline, which would be incorrect.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

File: docs/user-guide/sync_windows.md (hunk 1 above)
Comment (ishitasequeira) [reply docs/user-guide/sync_windows.md:24]: I agree with @vaniobegic here. @adriananeci , you would want to add/update the documentation as part of https://github.com/argoproj/argo-cd/pull/16846

**Pull request merged**